src/libstd/io/mod.rs

```diff
@@ -589,6 +1050,23 @@ pub trait BufRead: Read {
     /// return an error if the read bytes are not valid UTF-8. If an I/O error
     /// is encountered then `buf` may contain some bytes already read in the
     /// event that all data read so far was valid UTF-8.
+    ///
+    /// # Examples
+    ///
+    /// A locked standard input implements `BufRead`:
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::io::BufRead;
+    ///
+    /// let stdin = io::stdin();
+    /// let mut stdin = stdin.lock();
+    /// let mut buffer = String::new();
+    ///
+    /// stdin.read_line(&mut buffer).unwrap();
```

Comment:
Oh oops I think I mean `clear`. Basically this function will *append* to `buf`, so this is actually subtly incorrect:

```rust
while let Ok(n) = stdin.read_line(&mut buffer) {
    // work with `buffer`
}
```

The "correct loop" is:

```rust
while let Ok(n) = stdin.read_line(&mut buffer) {
    if n == 0 { break } // eof
    // work with `buffer`
    buffer.clear();
}
```